File: java/SwimmingCompetition/SwimmingCompetitionPersistence/src/main/java/swimmingcompetition/persistence/repository/ParticipantDBRepository.java
package swimmingcompetition.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import swimmingcompetition.model.Participant;
import swimmingcompetition.persistence.ParticipantRepository;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class ParticipantDBRepository implements ParticipantRepository {
    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger(OrganizerDBRepository.class);
    public ParticipantDBRepository(Properties properties) {
        this.dbUtils = new JdbcUtils(properties);
    }

    @Override
    public Participant findOne(Long aLong) {
        logger.traceEntry("finding participant with id {}", aLong);
        Connection connection = this.dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("select * from participant where id = ?")){
            preparedStatement.setLong(1, aLong);
            try(ResultSet resultSet = preparedStatement.executeQuery()){
                if(resultSet.next()){
                    Long id = resultSet.getLong("id");
                    String name = resultSet.getString("name");
                    int age = resultSet.getInt("age");
                    Participant participant = new Participant(name, age);
                    participant.setId(id);
                    return participant;
                }
            }
        }
        catch (SQLException ex){
            logger.error(ex);
            System.out.println("Eroare la DB -- Participant " + ex.getMessage().toString());
        }
        return null;
    }

    @Override
    public Iterable<Participant> findAll() {
        logger.traceEntry("finding all participants");
        Connection connection = dbUtils.getConnection();
        List<Participant> participantList = new ArrayList<>();
        try(PreparedStatement preparedStatement = connection.prepareStatement("select * from participant")) {
            try(ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    Long id = resultSet.getLong("id");
                    String name = resultSet.getString("name");
                    int age = resultSet.getInt("age");
                    Participant participant = new Participant(name, age);
                    participant.setId(id);
                    participantList.add(participant);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error la DB -- Participant " + e.getMessage().toString());
        }
        logger.traceExit(participantList);
        return participantList;
    }

    @Override
    public Participant save(Participant entity) {
        logger.traceEntry("saving participant {}", entity);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("insert into participant(name, age) values(?,?)", Statement.RETURN_GENERATED_KEYS)) {
            preparedStatement.setString(1, entity.getName());
            preparedStatement.setInt(2, entity.getAge());
            int result = preparedStatement.executeUpdate();
            ResultSet resultSet = preparedStatement.getGeneratedKeys();
            if(resultSet.next()){
                Long id = resultSet.getLong(1);
                entity.setId(id);
            }
            logger.trace("saved participant {}", entity);
            return entity;
        }
        catch (SQLException e) {
            logger.error(e);
            System.out.println("Error la DB -- Participant " + e.getMessage().toString());
        }
        return null;
    }

    @Override
    public void delete(Long aLong) {
        logger.traceEntry("deleting participant with id {}", aLong);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("delete from participant where id = ?")) {
            preparedStatement.setLong(1, aLong);
            int result = preparedStatement.executeUpdate();
            logger.trace("deleted participants {}", result);
        }
        catch (SQLException e) {
            logger.error(e);
            System.out.println("Error la DB -- Participant " + e.getMessage().toString());
        }
    }

    @Override
    public void update(Participant entity) {

    }
}

File: csharp/SwimmingCompetitionCS/persistence/ParticipantDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using log4net;
using model;

namespace persistence
{
    public class ParticipantDBRepository : ParticipantRepository
    {
        private static readonly ILog log = LogManager.GetLogger(typeof(ParticipantDBRepository));

        IDictionary<String, string> props;

        public ParticipantDBRepository(IDictionary<string, string> props)
        {
            this.props = props;
        }

        public Participant FindOne(long along)
        {
            log.InfoFormat("Finding participant with id {0}", along);
            IDbConnection connection = DBUtils.getConnection(props);
            using (var comm = connection.CreateCommand())
            {
                comm.CommandText = "select * from participant where id = @id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = along;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long id = dataR.GetInt32(0);
                        String name = dataR.GetString(1);
                        int age = dataR.GetInt16(2);
                        Participant participant = new Participant(name, age);
                        participant.Id = id;
                        return participant;
                    }
                }
            }
            log.InfoFormat("Finding NO participant with id {0}", along);
            return null;
        }

        public IEnumerable<Participant> FindAll()
        {
            log.Info("Finding all participants");
            IDbConnection connection = DBUtils.getConnection(props);
            IList<Participant> participants = new List<Participant>();
            using (var comm = connection.CreateCommand())
            {
                comm.CommandText = "select * from participant";
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        long id = dataR.GetInt32(0);
                        String name = dataR.GetString(1);
                        int age = dataR.GetInt16(2);
                        Participant participant = new Participant(name, age);
                        participant.Id = id;
                        participants.Add(participant);
                    }
                }
            }
            log.Info(participants);
            return participants;
        }

        public Participant Save(Participant entity)
        {
            log.InfoFormat("Saving participant {0}", entity);
            var connection = DBUtils.getConnection(props);
            using (var comm = connection.CreateCommand())
            {
                comm.CommandText = "insert into participant(name, age) values(@name,@age) RETURNING id";
                var paramName = comm.CreateParameter();
                paramName.ParameterName = "@name";
                paramName.Value = entity.Name;
                comm.Parameters.Add(paramName);
                
                var paramAge = comm.CreateParameter();
                paramAge.ParameterName = "@age";
                paramAge.Value = entity.Age;
                comm.Parameters.Add(paramAge);
                
                long generatedID = (long)comm.ExecuteScalar();

                entity.Id = generatedID;

                return entity;
            }
        }

        public void Delete(long id)
        {
            log.InfoFormat("Deleting participant with id {0}", id);
            IDbConnection connection = DBUtils.getConnection(props);
            using (var comm = connection.CreateCommand())
            {
                comm.CommandText = "delete from participant where id = @id";
                var idParam = comm.CreateParameter();
                idParam.ParameterName = "@id";
                idParam.Value = id;
                comm.Parameters.Add(idParam);
                var result = comm.ExecuteNonQuery();
                if(result == 1)
                    log.InfoFormat("Deleted participant with id {0}", id);
            }
        }

        public void Update(Participant entity)
        {
            // nu e nevoie
        }
    }
}